examples/chat-python-flask/static/chat.js

```diff
@@ -0,0 +1,35 @@
+const { hubURL, topic } = JSON.parse(document.getElementById('config').textContent)
+
+const subscribeURL = new URL(hubURL)
+subscribeURL.searchParams.append('topic', topic)
+
+const es = new EventSource(subscribeURL, { withCredentials: true })
+let ul = null
+es.onmessage = ({ data }) => {
+    const { username, message } = JSON.parse(data)
+    if (!username || !message) throw new Error('Invalid payload')
+
+    if (!ul) {
+        ul = document.createElement('ul')
+
+        const messages = document.getElementById('messages')
+        messages.innerHTML = ''
+        messages.append(ul)
+    }
+
+    const li = document.createElement('li')
+    li.append(document.createTextNode(`<${username}> ${message}`))
+    ul.append(li)
+}
+
+document.querySelector('form').onsubmit = function (e) {
+    e.preventDefault()
+
+    const body = new URLSearchParams({
+        data: JSON.stringify({ username: this.elements.username.value, message: this.elements.message.value }),
```

Comment:
Why do you need to send the username if the user is logged ?

Reply:
It is authorized to subscribe and to post to the, not really "logged in" (I mean it is authorized but not authenticated).
A proper login system should be added to the Python app (if appropriate), then the username could be set in the dedicated JWT claim, and sent along with the message. But then I think that it would be more efficient (and more secure) to send new messages to the Python app, that will then send it (including the current username) to the hub.